webrtc/src/data_channel/mod.rs

```diff
@@ -137,27 +137,35 @@ impl RTCDataChannel {
             let channel_type;
             let reliability_parameter;
 
-            if self.max_packet_lifetime == 0 && self.max_retransmits == 0 {
-                reliability_parameter = 0u32;
-                if self.ordered {
-                    channel_type = ChannelType::Reliable;
-                } else {
-                    channel_type = ChannelType::ReliableUnordered;
+            match (self.max_packet_lifetime, self.max_retransmits) {
+                (None, None) => {
+                    reliability_parameter = 0u32;
+                    if self.ordered {
+                        channel_type = ChannelType::Reliable;
+                    } else {
+                        channel_type = ChannelType::ReliableUnordered;
+                    }
                 }
-            } else if self.max_retransmits != 0 {
-                reliability_parameter = self.max_retransmits as u32;
-                if self.ordered {
-                    channel_type = ChannelType::PartialReliableRexmit;
-                } else {
-                    channel_type = ChannelType::PartialReliableRexmitUnordered;
+
+                (Some(max_packet_lifetime), None) => {
+                    reliability_parameter = max_packet_lifetime as u32;
+                    if self.ordered {
+                        channel_type = ChannelType::PartialReliableTimed;
+                    } else {
+                        channel_type = ChannelType::PartialReliableTimedUnordered;
+                    }
                 }
-            } else {
-                reliability_parameter = self.max_packet_lifetime as u32;
-                if self.ordered {
-                    channel_type = ChannelType::PartialReliableTimed;
-                } else {
-                    channel_type = ChannelType::PartialReliableTimedUnordered;
+
+                (None, Some(max_retransmits)) => {
+                    reliability_parameter = max_retransmits as u32;
+                    if self.ordered {
+                        channel_type = ChannelType::PartialReliableRexmit;
+                    } else {
+                        channel_type = ChannelType::PartialReliableRexmitUnordered;
+                    }
                 }
+
+                (Some(_), Some(_)) => unreachable!(),
```

Comment:
unreachable! is not acceptable here, since DataChannel library user may pass both  max_packet_lifetime and max_retransmits with values.

please keep the same logic as before, but with Option check.